### PR TITLE
Add repro for #5587

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Related to #5587

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs [hunk 1]
```diff
@@ -32,11 +32,11 @@ namespace SonarAnalyzer.Rules.CSharp
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     public sealed class ConditionalStructureSameImplementation : ConditionalStructureSameImplementationBase
     {
-        private static readonly DiagnosticDescriptor rule =
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs:35]: If you are already doing a cleanup of the classes, I think you could move the `DiagnosticDescriptor` and the `SupportedDiagnostics` to the base class, like [here](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/Rules/ConditionalStructureSameConditionBase.cs)

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
I know it is questionable if it is the scope of the PR or not, however I think once you work on a rule, all code smells should be cleaned if possible for that rule. (At least that is what I try to do (sometimes fail but always grow 😄 )) So, I suggest to fix the code smells also in the visual basic implementation of the rule, although you did not touch the visual basic test cases.

**Comment by costin-zaharia-sonarsource** [comment]
> So, I suggest to fix the code smells also in the visual basic implementation of the rule, although you did not touch the visual basic test cases.

I find this debatable. We should be able to take baby steps and clean up the code that we touch but not other files. Otherwise, when do you stop? There is always one more thing we can handle.

**Comment by csaba-sagi-sonarsource** [comment]
Fair point. That is why I said it is questionable. I try to limit this kind of PRs to a single rule, so we know that the whole rule is "clean". However, I also accept your stance. I will accept the PR.

**Pull request merged**